arch/ARM/STM32/drivers/dma_interrupts/stm32-dma-interrupts.ads

```diff
@@ -41,9 +41,16 @@ package STM32.DMA.Interrupts is
    is
       pragma Interrupt_Priority (Priority);
 
-      procedure Start_Transfer (Source      : Address;
-                                Destination : Address;
-                                Data_Count  : UInt16);
+      procedure Start_Transfer
+        (Source      : Address;
+         Destination : Address;
+         Data_Count  : UInt16);
+
+      procedure Start_Transfer_With_Interrupts
+        (Source             : Address;
+         Destination        : Address;
+         Data_Count         : UInt16;
+         Enabled_Interrupts : Interrupt_Selections := (others => True));
```

Comment:
I don't think that the selection of which interrupt is enabled should be decided by the user of this interface.

The service provided by this protected object is to perform the transfer and signal when it is completed (with the `Wait_For_Completion` entry). It's up to the implementation of the protected objet to decide which interrupts are required to achieve this service.

Reply:
I’ve made a much simpler change (#276) and will close this request.